release/nightly_tests/dataset/iter_batches_benchmark.py

```diff
@@ -0,0 +1,99 @@
+import sys
+from typing import Optional
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+def iter_batches(
+    ds: Dataset,
+    batch_size: Optional[int] = None,
+    batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
+    local_shuffle_buffer_size: Optional[int] = None,
+    use_default_params: bool = False,
+) -> Dataset:
+    num_batches = 0
+    if use_default_params:
+        for batch in ds.iter_batches():
+            num_batches += 1
+    else:
+        for batch in ds.iter_batches(
+            batch_format=batch_format,
+            batch_size=batch_size,
+            local_shuffle_buffer_size=local_shuffle_buffer_size,
+        ):
+            num_batches += 1
+    print("iter_batches done, num_rows:", ds.count(), "num_batches:", num_batches)
+    return ds
+
+
+def run_iter_batches_benchmark(benchmark: Benchmark):
+    ds = (
+        ray.data.read_parquet(
+            "s3://anonymous@air-example-data/ursa-labs-taxi-data/by_year/2018/01"
+        )
+        .repartition(24)
+        .fully_executed()
+    )
+
+    batch_formats = ["pandas", "numpy"]
+    batch_sizes = [128, 256, 512]
+
+    # Test default args.
+    test_name = "iter-batches-default"
+    benchmark.run(
+        test_name,
+        iter_batches,
+        ds=ds,
+        batch_format="INVALID",
```

Comment:
I assume this is just to confirm that `batch_format` is not passed through to `ds.iter_batches()`? Should we instead make the defaults for `iter_batches` to be `use_default_params=True` and `batch_format="INVALID"` (or just `batch_format=None`, since that should also raise an error) to make this a bit cleaner for the callers?